### PR TITLE
Actually Run All Unit Tests in CMake

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Seems the majority of our tests haven't been running for any platform that uses cmake (i.e. non windows). This rewrites the cmake to a more modern style for the tests and adds all the tests.

I had to disable the tests on any platform that didn't have modern cmake installed as I'm using the newer gtest integration.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Hmm for some reason its not seeing the replays

File: test/tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,198 +1,43 @@
...
-    "${ROOT_DIR}/src/openrct2/Version.cpp"
-    )
-
-# Create a re-usable library to save some compilation time
-add_library(test-common STATIC ${COMMON_TEST_SOURCES})
-set_target_properties(test-common PROPERTIES COMPILE_DEFINITIONS DISABLE_HTTP)
-
-# Setup testdata. It should be fine here, as the only way to reach here is by explicitly requesting tests.
-execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink "${CMAKE_CURRENT_LIST_DIR}/testdata" "${CMAKE_CURRENT_BINARY_DIR}/testdata")
-install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_DATADIR}/openrct2\" \"${CMAKE_CURRENT_BINARY_DIR}/data\")")
```
Comment (ZehMatt) [inline test/tests/CMakeLists.txt:32]: It probably doesn't find the data because of this

Comment (duncanspumpkin) [reply test/tests/CMakeLists.txt:32]: whats really fun is that it does find the data when i run it locally :)

**Comment by duncanspumpkin** [comment]
Finally worked out why the replays weren't working, we install the replays during install post build but gtest_discover_tests by default runs post build preinstall. i.e. when it discovers the tests there are no replays.

**Comment by janisozaur** [comment]
cmake 3.20+ update:

https://github.com/OpenRCT2/OpenRCT2/issues/15501#issuecomment-991978566 BSD is already up to date
https://github.com/OpenRCT2/OpenRCT2/issues/15501#issuecomment-992322863 flatpak builders are up to date

@gibmat @TheAssassin we are planning to up the required cmake version to 3.20 or later

**Comment by gibmat** [comment]
:+1: bookworm will be shipping with cmake 3.25 -- sounds good to me!

**Review by janisozaur — CHANGES_REQUESTED** [review]
`MultiLaunchTest` is now missing

**Comment by janisozaur** [comment]
Odd, I couldn't find it. Perhaps just some quirk about how the test results are presented preventing browser search from working

**Review by janisozaur — APPROVED** [review]

**Review by ZehMatt — APPROVED** [review]

**Pull request merged**